lib/terraform_landscape/helpers.rb

```diff
@@ -0,0 +1,54 @@
+# Provides Helpers for Project
+module TerraformLandscape::Helpers
+  class << self
+    def recursive_sort(obj)
+      case obj
+
+      when Array
+        recursive_sort_array_object(obj)
+
+      when Hash
+        recursive_sort_hash_object(obj)
+
+      else
+        obj
+
+      end
+    end
+
+    private
+
+    def recursive_sort_hash_object(obj)
+      Hash[
+        obj.map do |key, value|
+          [if key.respond_to? :sort
+             recursive_sort(key)
+           else
+             key
+           end,
+           if value.respond_to? :sort
+             recursive_sort(value)
+           else
+             value
+           end]
+        end.sort].extend HashDeepSortCompare
+    end
+
+    def recursive_sort_array_object(obj)
+      obj.map do |val|
+        if val.respond_to? :sort
+          recursive_sort(val)
+        else
+          val
+        end
+      end.sort
+    end
+  end
+
+  # Mixin Module for #recursive_sort
+  module HashDeepSortCompare
+    def <=>(other)
+      super(other) || to_a <=> other.to_a || to_s <=> other.to_s
+    end
+  end
+end
```

Comment:
Thanks for taking the time to get this ported without introducing a dependency. However, this seems like a lot of code for sorting what will be hashes or arrays (since it's JSON). You could accomplish the same thing with:

```
module TerraformLandscape::Helpers
  module_function

  def recursive_sort(obj)
    case obj
    when Array
      obj.map { |item| recursive_sort(item) }.sort
    when Hash
      obj.keys.sort.each_with_object({}) do |key, hash|
        hash[key] = recursive_sort(obj[key])
      end
    else
      obj
    end
  end
end
```

What do you think?

Reply:
I'm not against playing rubygolf all night, but I do have a life to live.  My goal was to write code that passed spec and lint, didn't introduce any new dependencies as requested, and satisfied this new functionality.  I also confirmed that it met my expectations in integration testing my terraform infra.  Thanks, but I think I'll just use my fork from here on out.  Feel free to further implement anon, etc. etc.

Reply:
I also worked to try at least to implement code that likely performed on all stable rubies, although CI doesn't exist to confirm that.